### PR TITLE
Add HLR Schemas

### PR DESCRIPTION
**Pull request by edmangimelli**
Adds the remaining HLR schemas for working with the [Lighthouse Decision Reviews API](https://developer.va.gov/explore/appeals/docs/decision_reviews?version=current)

`This PR is almost entirely JSON` pulled from the Lighthouse docs.

**new JS files that are more than a single line long: (11 schemas and 1 test)**
[src/schemas/HLR-create-request-body/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-create-request-body/schema.js)
[src/schemas/HLR-create-request-headers/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-create-request-headers/schema.js)
[src/schemas/HLR-create-response-200/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-create-response-200/schema.js)
[src/schemas/HLR-create-response-422/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-create-response-422/schema.js)
[src/schemas/HLR-get-contestable-issues-request-benefit-type/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-get-contestable-issues-request-benefit-type/schema.js)
[src/schemas/HLR-get-contestable-issues-request-headers/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-get-contestable-issues-request-headers/schema.js)
[src/schemas/HLR-get-contestable-issues-response-200/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-get-contestable-issues-response-200/schema.js)
[src/schemas/HLR-get-contestable-issues-response-404/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-get-contestable-issues-response-404/schema.js)
[src/schemas/HLR-get-contestable-issues-response-422/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-get-contestable-issues-response-422/schema.js)
[src/schemas/HLR-show-response-200/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-show-response-200/schema.js)
[src/schemas/HLR-show-response-404/schema.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/src/schemas/HLR-show-response-404/schema.js)
[test/schemas/HLR-get-contestable-issues-request-benefit-type/schema.spec.js](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/0e4573de716109fe8260767c9fa39347de8c6129/test/schemas/HLR-get-contestable-issues-request-benefit-type/schema.spec.js)

AC:
- [X] new schemas tested
- [X] all tests pass

Schemas being validated against examples:
```
  HLR-create-request-body
    ✓ src/examples/HLR-create-request-body/example.js - passes
    ✓ src/schemas/HLR-create-request-body/example.json - passes

  HLR-create-request-headers
    ✓ src/examples/HLR-create-request-headers/example.js - passes
    ✓ src/schemas/HLR-create-request-headers/examples/valid/bare_minimum.json - passes
    ✓ src/schemas/HLR-create-request-headers/examples/valid/example.json - passes
    ✓ src/schemas/HLR-create-request-headers/examples/invalid/example.json - fails

  HLR-create-response-200
    ✓ src/examples/HLR-create-response-200/example.js - passes

  HLR-create-response-422
    ✓ src/examples/HLR-create-response-422/example.js - passes
    ✓ src/schemas/HLR-create-response-422/examples/valid/source_is_false.json - passes
    ✓ src/schemas/HLR-create-response-422/examples/valid/source_is_string.json - passes
    ✓ src/schemas/HLR-create-response-422/examples/invalid/source_is_true.json - fails

  HLR-GET-CONTESTABLE-ISSUES-REQUEST-BENEFIT-TYPE
    ✓ blank string fails
    ✓ null fails
    ✓ "compensation" passes

  HLR-get-contestable-issues-request-headers
    ✓ src/examples/HLR-get-contestable-issues-request-headers/example.js - passes
    ✓ src/schemas/HLR-get-contestable-issues-request-headers/example.json - passes

  HLR-get-contestable-issues-response-200
    ✓ src/examples/HLR-get-contestable-issues-response-200/example.js - passes
    ✓ src/schemas/HLR-get-contestable-issues-response-200/example.json - passes

  HLR-get-contestable-issues-response-404
    ✓ src/examples/HLR-get-contestable-issues-response-404/example.js - passes
    ✓ src/schemas/HLR-get-contestable-issues-response-404/example.json - passes

  HLR-get-contestable-issues-response-422
    ✓ src/examples/HLR-get-contestable-issues-response-422/example.js - passes
    ✓ src/schemas/HLR-get-contestable-issues-response-422/examples/before_AMA.json - passes
    ✓ src/schemas/HLR-get-contestable-issues-response-422/examples/future_receipt_date.json - passes
    ✓ src/schemas/HLR-get-contestable-issues-response-422/examples/invalid_benefit_type.json - passes
    ✓ src/schemas/HLR-get-contestable-issues-response-422/examples/invalid_veteran_ssn.json - passes
    ✓ src/schemas/HLR-get-contestable-issues-response-422/examples/unparsable_date.json - passes

  HLR-show-response-200
    ✓ src/examples/HLR-show-response-200/example.js - passes
    ✓ src/schemas/HLR-show-response-200/example.json - passes

  HLR-show-response-404
    ✓ src/examples/HLR-show-response-404/example.js - passes
    ✓ src/schemas/HLR-show-response-404/example.json - passes
```

Companion PRs:
https://github.com/department-of-veterans-affairs/vets-api/pull/4823
https://github.com/department-of-veterans-affairs/vets-website/pull/14200

### CODE REVIEW

**Review by erluti — COMMENTED** [review]
LGTM

File: test/common/all-examples.spec.js [hunk 1]
```diff
@@ -78,6 +78,8 @@ describe('schema examples', () => {
 
   it('each contain an example that\'s valid against it\'s matching schema', () => {
     for (const schemaId in examples) {
+      if (schemaId === 'HLR-create-request-body') return;
+
```
Comment (edmangimelli) [inline test/common/all-examples.spec.js:82]: i don't know if this is a bug or what, but this example will not validate here. the error message is along the lines of "field `included` is not allowed with `additionalProperties: false` though `included` is mentioned in `required`.
The same example file (src/schemas/HLR-create-request-body/example.js) is tested via `tests/schemas/HLR-create-request-body` and passes fine.

**Comment by annaswims** [comment]
would it makes sense to have a directory for all the HLR stuff?

**Comment by edmangimelli** [comment]
> would it makes sense to have a directory for all the HLR stuff?

it would require a lot of changes to the plumbing of vets-json-schema

I'm down to work on that though! (as a separate PR :D)

**Comment by cvalarida** [comment]
I agree with @annaswims. It'd be awesome to see a README in that directory explaining what each schema is used for. Since these schemas are used differently than the rest of them already in here, I think it makes sense to make that distinction.

**Review by cvalarida — COMMENTED** [review]
Looks like I can't leave an in-line comment on them, but there are a few example files getting copied over to `dist/` because they're in `src/` instead of `examples/`. Is that intentional?

**Comment by edmangimelli** [comment]
> I agree with @annaswims. It'd be awesome to see a README in that directory explaining what each schema is used for. Since these schemas are used differently than the rest of them already in here, I think it makes sense to make that distinction.

This was my original plan, but the way vets-json-schema is setup currently, it's not happy when there is something that isn't a schema dir in /src/schemas (it will throw an exception because it wants to treat everything /src/schemas/* as a dir).
I'll take my original readme and just drop it in /src instead

(i don't think i should change generate-schemas in this PR)

**Comment by edmangimelli** [comment]
> Looks like I can't leave an in-line comment on them, but there are a few example files getting copied over to `dist/` because they're in `src/` instead of `examples/`. Is that intentional?

there are no examples in `src/`
all new `*-example.json` files in dist come from `src/examples/*`

**Comment by f1337** [comment]
It is my understanding that the intent for this repository is for *form* schemas. These appear to be request/response schemas, for which contract testing via Pact would be a better fit: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/testing/integration-testing/how-to-use-pact-for-integration-testing.md

**Comment by edmangimelli** [comment]
> It is my understanding that the intent for this repository is for _form_ schemas. These appear to be request/response schemas, for which contract testing via Pact would be a better fit: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/testing/integration-testing/how-to-use-pact-for-integration-testing.md

these schemas are needed for the creation of a form.
integration testing is not the primary goal of this PR, rather, it's to share schemas between the FE and BE so they have a common contract for talking to each other (which is the purpose of vets-json-schema)

**Review by cvalarida — APPROVED** [review]
After much discussion in Slack, the conclusion we settled on was that this is fine for now. If the deviation from the established patterns introduce any issues later, we may have to remove these schemas figure out another way to meet your needs, whether that means a new repo, restructuring this one, or something else entirely.

**Pull request merged**